flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -2604,6 +2604,9 @@ static void genOMP(
       std::get<parser::OmpClauseList>(simpleStandaloneConstruct.t), semaCtx);
   mlir::Location currentLocation = converter.genLocation(directive.source);
 
+  if (directive.v == llvm::omp::Directive::OMPD_taskwait && !clauses.empty())
+    TODO(converter.getCurrentLocation(), "taskwait with depend unsupported");
```

Comment:
taskwait could also have the nowait clause. I'm unsure whether that is supported yet